### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by ShanJiang929**
This branch implemented a superpixel model that is trained on ADNI dataset. The codebase include dependecy files and  python files for data preparation, model buidling, model training+validation_testing and model use example. The trained model and dataset is not included in the codebase. After 60 epochs of training, the model achieves mean PSNR of 28.82 with loss of 0.0013 for training; and mean PSNR of 27.56 for testing, which is higher than mean PSNR of arbitary lower resolution images (25.96).

### CODE REVIEW

**Comment by branme96** [comment]
This is an initial inspection - no action is required yet

Difficulty: Normal

General Comments: Nice approach overall, uses keras dataset functions that create a validation set. However, requires the user to manually extract AD and NC into one big folder. Code exists to compute metrics on the test set. Training plot looks good. Network architecture unchanged compared to provided Keras example.

[Recognition Problem]
"modules.py":

* Reimplementation of the Keras example without changes. 

"dataset.py":

* Reserves 20% of training dataset for validation.
* Based on readme and dataset code, student manually extracts AD and NC images into one big folder.
* Seems to be a nice implementation all round.

"train.py":

* Implements callbacks and early stopping to save best model.
* Computes metrics on whole test set after training (and loading best model). 
* Reports metrics by printing (average over dataset)

"predict.py":

* Loads model correct.
* Computes metrics on all images in the test set and prints image-by-image.
* Displays predictions and stores them.
* Computes the average performance over whole test set. Based on their divide by 10, it seems they only had 10 images in their test set.

"README.MD":

* Dependencies in separate text file (though it is referenced in the readme).
* Very brief and limited introduction into the model architecture.
* Reconstructed images in-line with other submissions.
* Provides usage information.
* Training/Validation plot looks good.

[Commit Log]
Commit Log looks ok.

**Comment by ShanJiang929** [comment]
Hello Dr. Shakes,

Thank you for your feedback. Could I please just double check that I do not need to make any change now since "This is an initial inspection - no action is required yet"? Following that, would there be a second round of feedback and if so when is the deadline to incorporate them? Thank you for your time.

Kind regards,
Shan
________________________________
发件人: branme96 ***@***.***>
发送时间: 2023年11月28日 11:14
收件人: shakes76/PatternAnalysis-2023 ***@***.***>
抄送: ikai ***@***.***>; State change ***@***.***>
主题: Re: [shakes76/PatternAnalysis-2023] Topic recognition (PR #157)


This is an initial inspection - no action is required yet

Difficulty: Normal

General Comments: Nice approach overall, uses keras dataset functions that create a validation set. However, requires the user to manually extract AD and NC into one big folder. Code exists to compute metrics on the test set. Training plot looks good. Network architecture unchanged compared to provided Keras example.

[Recognition Problem]
"modules.py":

  *   Reimplementation of the Keras example without changes.

"dataset.py":

  *   Reserves 20% of training dataset for validation.
  *   Based on readme and dataset code, student manually extracts AD and NC images into one big folder.
  *   Seems to be a nice implementation all round.

"train.py":

  *   Implements callbacks and early stopping to save best model.
  *   Computes metrics on whole test set after training (and loading best model).
  *   Reports metrics by printing (average over dataset)

"predict.py":

  *   Loads model correct.
  *   Computes metrics on all images in the test set and prints image-by-image.
  *   Displays predictions and stores them.
  *   Computes the average performance over whole test set. Based on their divide by 10, it seems they only had 10 images in their test set.

"README.MD":

  *   Dependencies in separate text file (though it is referenced in the readme).
  *   Very brief and limited introduction into the model architecture.
  *   Reconstructed images in-line with other submissions.
  *   Provides usage information.
  *   Training/Validation plot looks good.

[Commit Log]
Commit Log looks ok.

―

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used
## Documentation
ReadMe acceptable, no problem statement -1
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in InCorrect Branch) -2
No Feedback required
Request Description minimal -1

**Comment by shakes76** [comment]
Wrong branch used, please update branch to correct one and ensure repo READMEs are restored. Does not affect grade only the merge of your PR.
